### ISSUE TITLE
GHA failing

### ISSUE THREAD

**Issue by mattsecrest** [r-universe/genentech#1]
@jeroen can we please turn off GHA for the this `r-universe` mirror? It is failing and troubleshooting not needed as we have our own CICD pipeline on the main repo.

**Comment by jeroen** [comment]
You can remove the package from your runiverse via your registry: https://github.com/Genentech/genentech.r-universe.dev

**Comment by mattsecrest** [comment]
Thanks @jeroen so there's no way to keep on R-universe and as a repo where people can download while ignoring these GHA?

**Comment by jeroen** [comment]
Not sure what you mean. R-universe uses this workflow to build all the material that is deployed on https://genentech.r-universe.dev. So if you disable CI/CD the workflow, your package won't be built and deployed anymore.

I see that your vignette currently fails to build due to a problem with rcmdstan? Do we need to install something on the build image to make that work?

**Comment by cicdguy** [comment]
@jeroen - Yes it'll be nice to have STAN and `cmdstanr` installed on the Universe container images. There are other packages that will also leverage STAN (from other organizations).

tl;dr:

```R
install.packages(
 "cmdstanr",
  repos = c(
    "https://cloud.r-project.org/",
    "https://mc-stan.org/r-packages/"
  ),
  Ncpus = parallel::detectCores(),
  ask = FALSE
)

# Install cmdstan
cmdstanr::install_cmdstan(cores = parallel::detectCores())
```

Also nice to have the following global env vars set for STAN to be located:

```bash
CMDSTAN=~/.cmdstan
CMDSTAN_PATH=~/.cmdstan
```

**Comment by gravesti** [comment]
@jeroen Yeah, the package relies on the R package `cmdstanr` which provides an interface to the `CmdStan` utilities (https://mc-stan.org/users/interfaces/cmdstan.html). Without those command line tools the package can't work, but I'm not sure if we should be downloading and compiling them in our packages installation.

Normally we would expect users to install `cmdstanr` and then use its R functions to download and compile `CmdStan`. If CmdStan were already installed on the image and can be detected by `cmdstanr` that would be a big win for us.

We are still trying to understand how we can do this in a CRAN compatible way.

**Comment by mattsecrest** [comment]
@jeroen would the best way to add `cmdstanr` be to throw it in [base image](https://github.com/r-universe-org/base-image)?

The install is a [bit annoying ](https://mc-stan.org/cmdstanr/) but happy to try a PR to on the base image to see if this helps.

**Comment by jeroen** [comment]
Sorry for dropping the ball. Yes we can add external utilities to the base image. Is there a way to install the required utilities without preinstalling the R packages on the base image? 

We prefer not to preinstall R packages on the base image because they drag in a lot of dependencies and they tend to get outdated quickly. So preferably we would only preinstall a minimal amount of required stan utilities on the base image, and install R packages on the fly during the CI workflow. Does that make sense?

**Comment by mattsecrest** [comment]
Hey @jeroen ,

Yes we can add the required utilities (`CmdStan`) and I just got this working, but I timed it at ~6 minutes using multiple threads. We could then download the R package (`cmdstanr`) via remotes (an additional ~4 minutes).

What do you think? You can see the base image updates here: https://github.com/mattsecrest/runiverse-base-image/pull/1/files

**Comment by mattsecrest** [comment]
(Just to be clear, no need to install `cmdstanr` on the base image :) )

**Comment by jeroen** [comment]
The problem is not the build time, but the size of the base image. The base image gets pulled in for every CI job, thousands of times per day. Your PR makes the base image 1gb larger, so it adds like a minute of build time to all builds in r-universe.

Would it be possible to update the PR such that after running `make build` your remove all the files inside `cmdstan-2.33.1 ` except for those that are needed at runtime by cmdstanr?

**Comment by cicdguy** [comment]
Thanks a bunch @jeroen!

**Comment by mattsecrest** [comment]
Thanks so much @jeroen !

**Comment by jeroen** [comment]
No problem. I'm also trying to make it work on Mac and Windows, but I can't find any precompiled cmdstan releases for Win/Mac. The `collab-cmdstan-2.33.1.tgz` releases that I use on Linux don't seem to work on Windows and Mac: https://github.com/r-universe/genentech/actions/runs/6710804435/job/18239174915

Is it really expected that all users on Win and Mac install cmdstan from source?
